lib/elixir/lib/kernel/typespec.ex

```diff
@@ -192,20 +192,65 @@ defmodule Kernel.Typespec do
 
   @doc false
   def translate_typespecs_for_module(_set, bag) do
-    types = Enum.map(take_typespec(bag, :type), &translate_type/1)
-    specs = Enum.map(take_typespec(bag, :spec), &translate_spec/1)
-    callbacks = Enum.map(take_typespec(bag, :callback), &translate_spec/1)
-    macrocallbacks = Enum.map(take_typespec(bag, :macrocallback), &translate_spec/1)
+    type_typespecs = take_typespec(bag, :type)
+    defined_type_pairs = collect_defined_type_pairs(type_typespecs)
+    state = %{defined_type_pairs: defined_type_pairs, used_type_pairs: []}
```

Comment:
I quite not like the name `state`, but it the best name I came up with 😅 

I also disliked that `:defined_type_pairs` and `:used_type_pairs` have different structures but the same suffixes, but that's the minimum info that is required. I guess not a big deal 🤷‍♂️